src/NuGet.Core/NuGet.Commands/RestoreCommand/SourceRepositoryDependencyProvider.cs

```diff
@@ -239,9 +241,28 @@ public class SourceRepositoryDependencyProvider : IRemoteDependencyProvider
                     };
                 }
             }
-            catch(FatalProtocolException e) when(_ignoreFailedSources)
+            catch (FatalProtocolException e)
             {
-                await LogWarningAsync(libraryRange.Name, e);
+                if (_ignoreFailedSources)
+                {
+                    await LogWarningAsync(libraryRange.Name, e);
+                }
+                else
+                {
+                    NuGetLogCode logCode = e.LogCode;
+
+                    RestoreLogMessage logMessage = null;
+                    if (e.InnerException != null)
+                    {
+                        logMessage = RestoreLogMessage.CreateError(logCode, e.Message + Environment.NewLine + e.InnerException.Message);
```

Comment:
Consider making a utility method to extract the Message from an `Exception` so this logic doesn't have to get duplicated.
Also, either use $"{0}" style strings String.Format instead of concatenation.

Reply:
Fixed in next iteration.